### ISSUE TITLE
Some deleted keys might appear after compaction

### ISSUE THREAD

**Issue by zhangjinpeng1987** [facebook/rocksdb#2722]
We encounter a strange problem, some keys have been deleted might appear after compaction. These keys are continuous.
I dump the manifest, and find the sst file(307784) that contains these keys.
```
 {"Level": 2, "FileNumber": 307784, "FileSize": 35099575, "SmallestIKey": "'7A7480000000000000FF2D5F728000000006FF797A5E0000000000FAFA88E755E593FFEA' seq:0, type:1", "LargestIKey": "'7A7480000000000000FF2D5F728000000006FF8F8C590000000000FAFA88E5E1EE23FFD6' seq:0, type:1"}
```
And then, I dump the sst file, and make a check if these keys are exist in this sst file, the answer is yes. And I check the LOG to find out when this sst file has been created.
```
2017/08/10-05:11:59.737441 7f8b4fdfc700 [db/compaction_job.cc:1118] [default] [JOB 32585] Generated table #307784: 1182756 keys, 35099575 bytes
```
When our consistency check program run at about `2017/08/10-05:20:`, we found these unexpected keys.

**Comment by siddontang** [comment]
Btw, When I try to repair the db, I meet the coredump, see #2712

**Comment by zhangjinpeng1987** [comment]
This [pr](https://github.com/facebook/rocksdb/pull/2726) will fix it.

**Comment by yiwu-arbug** [comment]
Fix included in 5.7.1.
